### PR TITLE
Support retrieve azure credentials from AzureClusterIdentity

### LINKED ISSUE #4429: Support retrieve azure credentials from AzureClusterIdentity
**Bug description**
Related to [TKG-17286](https://jira.eng.vmware.com/browse/TKG-17286)
Since cluster-api-for-azure [release-0.5.0](https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/tag/v0.5.0), AzureClusterIdentity has become the recommended approach for cluster creation. And set credential environment variables method is deprecated. 
However, for TKGm side, we still require user to provide credential environment variables when create workload cluster. This is not expected. We need to keep consistent with upstream.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- (x) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by xiujuanx**
### Which issue(s) this PR fixes

Fixes #4429 

### Release note

```release-note

```

### CODE REVIEW

File: tkg/clusterclient/credentials.go [hunk 1]
```diff
@@ -443,6 +443,43 @@ func (c *client) GetAzureCredentialsFromSecret() (azureclient.Credentials, error
...
 
+func (c *client) GetAzureCredentialsFromIdentity(identityName string, identityNamespace string) (azureclient.Credentials, error) {
+	var (
+		clientSecretName      string
+		clientSecretNamespace string
+	)
+	res := azureclient.Credentials{}
+
+	azureClusterIdentityList := &capzv1beta1.AzureClusterIdentityList{}
+	err := c.ListResources(azureClusterIdentityList, &crtclient.ListOptions{})
```
Comment (hongtaiwang) [inline tkg/clusterclient/credentials.go:454]: is there any methods that could filter the listed resources during `ListResources()` api call (like within `&crtclient.ListOptions{}` )? Not sure if there will be timing issues if a large amount of cluster identity exists.

Comment (xiujuanx) [reply tkg/clusterclient/credentials.go]: Thanks @hongtaiwang ! I switch this behavior from list identity resource to get identity resource, now!

**Comment by weiwenli97** [comment]
lgtm

File: tkg/clusterclient/credentials.go [hunk 2]
```diff
@@ -443,6 +443,37 @@ func (c *client) GetAzureCredentialsFromSecret() (azureclient.Credentials, error
 	return res, nil
 }
 
+func (c *client) GetAzureCredentialsFromIdentity(identityName string, identityNamespace string) (azureclient.Credentials, error) {
+	var (
```
Comment (wjun) [inline tkg/clusterclient/credentials.go:447]: Declare these two variables directly when we use them?

File: tkg/clusterclient/credentials.go [hunk 3]
```diff
@@ -443,6 +443,37 @@ func (c *client) GetAzureCredentialsFromSecret() (azureclient.Credentials, error
...
+	clientSecretNamespace = azureClusterIdentityGet.Spec.ClientSecret.Namespace
+
+	if res.ClientID == "" || res.TenantID == "" || clientSecretName == "" || clientSecretNamespace == "" {
+		return res, errors.New("unable to retrieve azure credentials from AzureClusterIdentity")
+	}
+
+	secretGet := &corev1.Secret{}
+	err = c.GetResource(secretGet, clientSecretName, clientSecretNamespace, nil, nil)
+	if err != nil {
+		return res, errors.Wrap(err, "unable to retrieve secret is referenced by AzureClusterIdentity")
```
Comment (wjun) [inline tkg/clusterclient/credentials.go:470]: Better to rephrase the message as: unable to retrieve secret that is referenced by AzureClusterIdentity

Comment (xiujuanx) [reply tkg/clusterclient/credentials.go]: Thanks @wjun ! Have changed as your suggestions.

File: tkg/clusterclient/credentials.go (hunk 2 above)
Comment (xiujuanx) [reply tkg/clusterclient/credentials.go]: Done!

**Review by wjun — APPROVED** [review]
LGTM and merge after passing capz cc pipeline.

**Pull request merged**